### PR TITLE
Tanks have tiers

### PR DESCRIPTION
**Pull request by Ghilker**
## About The Pull Request
This PR changes how tanks work, giving them 4 tiers. What changes is the breaking point (along with the minimum leaking point) and the integrity of each tank
-tier 1: leaking point = 1064 Kpa, integrity = 1
-tier 2: leaking point = 2280 Kpa, integrity = 3
-tier 3: leaking point = 3040 Kpa, integrity = 5
-tier 4: leaking point = 9119 Kpa, integrity = 7
All the tanks in the station are tier 1 with an internal pressure of 1035 Kpa, so all of those are safe to use.
The rupture point is scaled too, so higher pressure will make more powerful explosions (t4 tank are locked under weaponry but i don't know if is a good spot, maybe advanced weaponry?)
This PR also fixes canisters/pumps/scrubbers runtiming if the tank inside them break/explode/get deleted.

## Why It's Good For The Game
Making powerful bombs will need a bit of research before being obtainable, toxins will have more work to do instead of 10/15 minute 50k research bomb and then close the toxin department.

## Changelog
:cl:
add: tiered tanks, they only differ for the breaking point and the integrity, no volume has been touched
fix: fix canisters/pumps/scrubbers runtiming when a tank inside them broke or gets deleted, blocking the machine for some time
/:cl:

### CODE REVIEW

**Comment by Ghilker** [comment]
@Barhandar fastest disliker on the west coast

**Comment by MrDoomBringer** [comment]
seems like this is just adds another time gate to big research bombs instead of requiring more skill / departmental cooperation

**Comment by Tlaltecuhtli** [comment]
> seems like this is just adds another time gate to big research bombs instead of requiring more skill / departmental cooperation

which one of these is needed to follow a youtube tutorial

**Comment by MrDoomBringer** [comment]
@Tlaltecuhtli while the current bomb-making process is indeed youtube-tutorial-tier, I'm not saying it can't be improved - just that this might not be the best way to do it

**Comment by trerri** [comment]
I like this, it's basically the only toxins rework we have atm, and you can make a small bomb with a small tank, then a larger one with a bigger tank to get points.

**Comment by wesoda25** [comment]
Bomb making process will always be tutorial tier unless the reactions are randomized each round, so that some experimentation is required to find the perfect ratio. I don't think it's a big issue to make 50k bombs time gated.

**Comment by itseasytosee** [comment]
This isn't even interdepartmental-based waiting, its just waiting. Emergent gameplay loop if you ask me.

**Comment by Tlaltecuhtli** [comment]
> Also how the hell does research bigger tanks make sense? We can make this tank, but we don't yet have the technological understanding to make it bigger.

because all tech is protected by copyrights and scientists use literal bitcoins to buy said techs

**Comment by nicbn** [comment]
This is a nerf for atmos as well, I don't think this is needed rn.

**Comment by trerri** [comment]
@itseasytosee did you even read the pr the tank isnt bigger its just made of plasteel and bluespace crystal

File: code/modules/research/designs/autolathe_designs.dm [hunk 1]
```diff
@@ -290,14 +290,32 @@
...
+	build_type = AUTOLATHE | PROTOLATHE
+	materials = list(/datum/material/iron = 1000, /datum/material/plasma = 1000)
+	build_path = /obj/item/tank/tier_2
+	category = list("Misc","Equipment")
+	departmental_flags = DEPARTMENTAL_FLAG_ENGINEERING | DEPARTMENTAL_FLAG_CARGO | DEPARTMENTAL_FLAG_SCIENCE
+
+/datum/design/generic_gas_tank_t3
+	name = "Generic Tier 3 Gas Tank"
+	id = "generic_tank_t3"
+	build_type = AUTOLATHE | PROTOLATHE
```
Comment (ATH1909) [inline code/modules/research/designs/autolathe_designs.dm:314]: wait, if you can just autolathe these, what's the point of locking them behind a tech node?

**Comment by ATH1909** [comment]
image

**Comment by bobbahbrown** [comment]
you should check with oranges and see if this falls into where he would like timing to be with research

File: code/modules/research/designs/autolathe_designs.dm (hunk 1 above)
Comment (Ghilker) [reply code/modules/research/designs/autolathe_designs.dm:314]: I'll fix that

**Comment by Ghilker** [comment]
This commit adds another tier, the bluespace one has the same rupture point of the old one, shifting the t2 and t1 lower on the chart. Check the PR body for the update

**Comment by LemonInTheDark** [comment]
How does this effect creating cores? It's already a part of toxins, and this seems like it might discourage it. Also I need to test this with single caps, brb

**Comment by Ghilker** [comment]
Well to create cores you have to use good bombs right? Now you have to research "better bombs" before being able to create cores

**Comment by trerri** [comment]
cores are meant to be as braindead easy as possible to make though

**Comment by Hamcha** [comment]
Speedrunning toxins is the fun part of toxins (imo), timegating will only make toxins boring. Toxins is a wait-heavy job: wrench this can, wait, unwrench, wrench another can, wait, etc.

The fun myself (and I would assume Kel) have found ourselves was to optimize the waiting to be as little as possible (ie. speedrunning it), therefore, I see an artificial timegate as like, an unskippable 15 minutes cutscene before you get to play your 5 minutes of game, is it really worth it?

> Well to create cores you have to use good bombs right? Now you have to research "better bombs" before being able to create cores

I honestly haven't seen how little you really need to make cores, I made some really bad bombs (not emptying oxy/plasma tanks out of the dispenser, only filling them halfway, using only hot plasma etc) and they still got over the core production threshold by a large margin.
Plus, there is currently zero appreciation for non-bluespace cores right now (at least on Terry), so making them feels like useless busywork after you maxcap.

**Comment by LemonInTheDark** [comment]
If toxins is currently built around making ttvs to detonate and make cores with/generate research points, does giving them a time gate here improve things at all? Do they have other stuff to do at the time, since they're just waiting for the right tech before they can do their job.

**Comment by TJatPBnJ** [comment]
good

**Comment by LemonInTheDark** [comment]
I think this change is better suited to a larger toxins revamp, and I don't trust some of the numbers here. If you're interested in getting after something like that, please write up your thoughts in a design doc so I can understand the flow better.

**Pull request closed without merging**